tools/test/build_api/build_api_test.py

```diff
@@ -238,5 +240,41 @@ def test_build_library_no_app_config(self, mock_prepare_toolchain, mock_exists,
         self.assertEqual(args[1]['app_config'], None,
                          "prepare_toolchain was called with an incorrect app_config")
 
+    @patch('tools.build_api.Resources')
+    @patch('tools.build_api.mkdir')
+    @patch('os.path.exists')
+    @patch('tools.build_api.prepare_toolchain')
+    def test_merge_region_no_fit(self, mock_prepare_toolchain, mock_exists, _, __):
+        """
+        Test that merge region fails as expected when part size overflows region size.
+        """
+        with patch("tools.build_api.intelhex_offset") as _intelhex_offset:
```

Comment:
Could you move this patch into the `@patch` directorator section above the funnction?